python/mead/downloader.py

```diff
@@ -0,0 +1,300 @@
+import sys
+
+def get_file(file_name):
+    import shutil
+    import os
+    if os.path.exists(file_name):
+        shutil.rmtree(file_name)
+    return file_name
+
+
+def extract_gzip(file_loc):
+    import gzip
+    import shutil
+    temp_file = get_file("{}.1".format(file_loc))
+    with gzip.open(file_loc, 'rb') as f_in:
+        with open(temp_file, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+    if mime_type(temp_file) == "application/x-tar":
+        return extract_tar(temp_file)
+    else:
+        shutil.move(temp_file, file_loc)
+        return file_loc
+
+
+def extract_tar(file_loc):
+    import tarfile
+    import os
+    temp_file = get_file("{}.1".format(file_loc))
+    with tarfile.open(file_loc, "r") as tar_ref:
+        tar_ref.extractall(temp_file)
+    if len(os.listdir(temp_file)) != 1:
+        raise RuntimeError("tar extraction unsuccessful")
+    return os.path.join(temp_file, os.listdir(temp_file)[0])
+
+
+def extract_zip(file_loc):
+    import zipfile
+    temp_file = get_file("{}.1".format(file_loc))
+    with zipfile.ZipFile(file_loc, "r") as zip_ref:
+        zip_ref.extractall(temp_file)
+    return temp_file
+
+
+def mime_type(loc):
+    import magic
+    return magic.Magic(mime=True).from_file(loc)
+
+
+def get_cache_dir():
+    import json
+    try:
+        data_cache = json.load(open("config.json")).get("datacache", "~/.bl-dataset-embeddings/")
+    except IOError:
+        data_cache = "~/.bl-dataset-embeddings/"
```

Comment:
why not check for the file before loading it instead of silently recovering from an error?